sygnal/apnspushkin.py

```diff
@@ -351,7 +351,9 @@ def _get_payload_event_id_only(
         payload = {}
 
         if device.data:
-            payload.update(device.data.get("default_payload", {}))
+            default_payload = device.data.get("default_payload")
+            if isinstance(default_payload, dict):
+                payload.update(default_payload)
```

Comment:
This change makes us ignore any `default_payload`s which aren't a dict. Is providing such a `default_payload` legitimate, or should we consider it a request error (e.g. HTTP 400 Bad Request)?

The same for gcm pushkin below.

Reply:
Note that the webpush pushking already does this exact code.

Reply:
The Sygnal docs do specify that `default_payload` should be a dict, (although I can't find anything about `default_payload` in the spec fwiw): https://github.com/matrix-org/sygnal#pusher-data-configuration

I think it's fair to consider anything other than a dict a Bad Request and throw an error in those cases, if people agree with this I can also change the behavior of the webpush pushkin to reflect that as well.

Reply:
If we decide to go down that route, I'd rather we reject the pushkey (tell the homeserver to forget that pushkey+device data combination) rather than 400. Otherwise the homeserver will continue to send us notifications that we refuse to handle.

Reply:
It sounds like some of the options to handle `default_payloads` of the incorrect type are:
-throw a bad request error and 400 
-reject the pushkey
-drop the `default_payload` data silently (what's currently happening in the PR)

I'm trying to think about what would make the most sense/be least annoying to handle for the end user. Would it make sense to ignore the `default_payload` if it is malformed but warn about that in the logs?

Reply:
An invalid `default_payload` is a programming error, I believe, so rejecting the `pushkey` probably makes the most sense?

The push spec is a bit confusing since much of it is left up to implementation details between client app and push server (with the homeserver just proxying between).

Reply:
Alright I will reject the `pushkey`, thanks for everyone's input!